### ISSUE TITLE
Autopage tags not filtering on specified collection

### ISSUE THREAD

**Issue by Markyparky56** [sverrirs/jekyll-paginate-v2#107]
Hi, I'm still pretty new to Jekyll so all the different terminologies still seem a bit confusing to me. I've been trying to set up a jekyll site which is split into three sections (food, projects & journal). So far I have managed to make this work through the use of collections:

_from \_config.yaml_
```
collections:
  food:
    output: true
    permalink: /food/:path/
  projects:
    output: true
    permalink: /projects/:path/
  journal:
    output: true
    permalink: /journal/:path/
```

I have three "top-level" pages which paginate for these three collections, for example the front matter on the Project.md page looks like:
```
---
layout: home
title: Projects
pagination:
  enabled: true
  collection: projects  
  per_page: 5
  permalink: /:num/
  sort_field: 'date'
  sort_reverse: true
---
```
This works fine for my purposes.

I also have want to allow for arbitrary tags on each post. Paginate seems to be able to cope with this, via autopage (as seen in the 3rd example site), and so I have a `autopage_tags.html` which spits out a set of pages for all the different tags. 
The front matter of which looks like:
``` 
---
layout: default
autopages:
  permalink: /tags/:tag
pagination:
  enabled: true
  collection: all
  debug: false
---
```

However when I tried to tweak the front matter to create sub-groups for each collection it doesn't appear to be working. Instead I just get the same set of pages again, in the specified sub-folder.
The front matter for my `autopage_tags_projects.html` looks like:
```
---
layout: default
autopages:
  permalink: /tags/projects/:tag
pagination:
  enabled: true
  collection: projects
  debug: false
---
```
Note that the `pagination->collection` field is now just `projects` and not `all`. As I understand it `all` is supposed to be a special case which includes all collections, however specifying a collection by name doesn't limit paginate to that collection. 

Is this a bug, or am I misusing paginate, jekyll, collections or something entirely different?

**Comment by Markyparky56** [comment]
Likely I am just misusing/abusing collections. Added `category` fields to the front matter of all the (dummy) posts then added filters for the specific categories to the autopage layout front matters and it seems to be happy filtering them that way.

**Comment by mark-making** [comment]
> Likely I am just misusing/abusing collections. Added `category` fields to the front matter of all the (dummy) posts then added filters for the specific categories to the autopage layout front matters and it seems to be happy filtering them that way.

Would you mind showing your solution i.e. how you filtered the specific categories on the auto layout pages please? Thanks. 😄

**Comment by Markyparky56** [comment]
@mark-making Not sure exactly what you're asking for, so here's a run down of my setup

The relevant parts of _config.yml:
```yaml
permalink: pretty

# Build settings
highlighter: rouge
markdown: kramdown
theme: minima
plugins:
  - jekyll-paginate-v2

collections:
  food:
    output: true
    permalink: /food/:path/
  projects:
    output: true
    permalink: /projects/:path/
  journal:
    output: true
    permalink: /journal/:path/
  portfolioItems:
    output: false
    permalink: /portfolio/:path/

autopages:
  enabled: true
  tags: 
    layouts: 
      - autopage_tags.html # For all collections 
      - autopage_tags_food.html 
      - autopage_tags_journal.html
      - autopage_tags_projects.html
    enabled: true
  collections:
    enabled: false
  categories:
    enabled: false

pagination: 
  enabled: true
  per_page: 5
  permalink: '/:num/'
  title: ':title | :num of :max'
  limit: 0
  sort_field: 'date'
  sort_reverse: true
  trail:
    before: 2
    after: 2
```
I then have a hardcoded markdown page for each category index with the frontmatter in the form of:

```yaml
---
layout: section_home
title: Food
permalink: /food/
pagination:
  enabled: true
  collection: food  
  per_page: 5
  permalink: /:num/
  sort_field: 'date'
  sort_reverse: true
---
```
The layout `section_home` looks like:

```html
---
layout: default
---
<!-- Just some nice to have styles for the pager buttons -->
<style>
  ul.pager { text-align: center; list-style: none; }
  ul.pager li {display: inline;border: 1px solid black; padding: 10px; margin: 5px;}
  .selected{ background-color: magenta; }
</style>

<div class="home">

  <h1 class="page-heading">Latest Posts</h1>
  
  {{ content | markdownify }}

  <ul class="post-list">
    <!-- 
        Here is the main paginator logic called.
        All calls to site.posts should be replaced by paginator.posts 
    -->
    {% for post in paginator.posts %}
      <li>
        <span class="post-meta">
          {{ post.date | date: "%b %-d, %Y" }}{% if post.collection %} | {{ post.collection }}{% endif %}
          {% if post.tags %}
            <br>
            Tags: 
            {% for tag in post.tags %}
              <a href="/tags/{{ tag }}">{{ tag }}</a>
            {% endfor %}
          {% endif %}
        </span>
        <h2>
          <a class="post-link" href="{{ post.url | relative_url }}">{{ post.title | escape }}</a>
        </h2>
      </li>
    {% endfor %}
  </ul>
  
  <!-- 
    Showing buttons to move to the next and to the previous list of posts (pager buttons).
  -->
  {% if paginator.total_pages > 1 %}
  <ul class="pager">
      {% if paginator.first_page %}
      <li class="previous">
          <a href="{{ paginator.first_page_path | prepend: site.baseurl | replace: '//', '/' }}">First</a>
      </li>
      {% endif %}

      {% if paginator.previous_page %}
      <li class="previous">
          <a href="{{ paginator.previous_page_path | prepend: site.baseurl | replace: '//', '/' }}">&larr; Newer Posts</a>
      </li>
      {% endif %}

      {% if paginator.page_trail %}
        {% for trail in paginator.page_trail %}
          <li {% if page.url == trail.path %}class="selected"{% endif %}>
              <a href="{{ trail.path | prepend: site.baseurl | replace: '//', '/' }}" title="{{trail.title}}">{{ trail.num }}</a>
          </li>
        {% endfor %}
      {% endif %}

      {% if paginator.next_page %}
      <li class="next">
          <a href="{{ paginator.next_page_path | prepend: site.baseurl | replace: '//', '/' }}">Older Posts &rarr;</a>
      </li>
      {% endif %}

       {% if paginator.last_page %}
      <li class="previous">
          <a href="{{ paginator.last_page_path | prepend: site.baseurl | replace: '//', '/' }}">Last</a>
      </li>
      {% endif %}
  </ul>
  {% endif %}

  <!--<p class="rss-subscribe">subscribe <a href="{{ "/feed.xml" | relative_url }}">via RSS</a></p>-->

</div>
```


This creates paginations of all the post category/collections. It's built off of the standard pagination with buttons, I'll get around to styling it properly eventually but it's functional.

Each post is placed in the relevant collection folder (i.e. _journal, _food, _projects) with frontmatter in the form of: 

```yaml
---
layout: post
title: "Honours Project Week 4 Retrospective"
date: 2018-10-12 14:15:00 +0100
collection: journal
category: journal
tags: honours-project
---
```

And then the autopage tags use frontmatter:
```yaml
---
layout: default
autopages:
  permalink: /tags/food/:tag
pagination:
  enabled: true
  collection: food
  category: food
  debug: false
---
```

Note that each post belongs to an identical collection and category, and the autopage paginations on both collection and category. It's been a while since I set this up so I've forgotten a lot of what I did to make this work but that appears to be what makes it tick. 

The autopage_tags_*.html pages are all pretty similar to the section_home layout paginatior set up with a custom page-heading. 

I also jerry-rigged a rather obscene amount of liquid code to generate a tags-list page which can be found [here](https://gist.github.com/Markyparky56/f692b278f46219d3a7f5a6ffb8dee416). It's not fully generic and requires knowledge of each category/collection but is able to collect, count and construct a tag list for each collection.

**Comment by mark-making** [comment]
@Markyparky56 Thanks for your response. I'm having issues with the categories being output in two separate collections, when I want to filter some categories specifically to a collection using the autopage plugin. But not working.
